File: CS50s Introduction to Computer Science/pset1/2. Mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int maxSize = 8;
    int size = 0;

    // Ask the user to input a pyramid size, and continue to do so until it's valid
    do
    {
        size = get_int("Please enter a number between 1 and 8: \n");
    }
    while (size < 1 || size > 8);

    int row = 1;
    for (int i = 0; i < size; i++)
    {
        // First print out the spaces on the row
        for (int j = 0; j < size - row; j++)
        {
            printf(" ");
        }
        // Then print out the hash marks for the left pyramid
        for (int k = 0; k < row; k++)
        {
            printf("#");
        }
        // Then print the two spaces
        for (int l = 0; l < 2; l++)
        {
            printf(" ");
        }
        // Lastly print out the hash marks for the right pyramid
        for (int m = 0; m < row; m++)
        {
            printf("#");
        }
        row++;
        printf("\n");
    }
}

File: CS50s Introduction to Computer Science/pset6/mario/more/mario.py
from cs50 import get_int

while True:
    size = get_int('Please enter a size between 1 - 8:\n')
    if size < 0 or size > 8:
        size = input('Please enter a size between 1 - 8:\n')
    else:
        break
row = 1
for i in range(size):
    # First print out the spaces on the row
    for j in range(size - row):
        print(" ", end = "")
    # Then print out the hash marks for the left pyramid
    for k in range(row):
        print("#", end = "")
    # Then print the two spaces
    for i in [1, 2]:
        print(" ", end = "")
    # Lastly print out the hash marks for the right pyramid
    for m in range(row):
        print("#", end = "")
    row += 1
    print("\n", end = "")

